javaslang/src/main/java/javaslang/collection/JavaConverters.java

```diff
@@ -0,0 +1,577 @@
+            return delegate.get(index);
+        }
+
+        @Override
+        public int indexOf(Object obj) {
+            return narrowed().indexOf(obj);
+        }
+
+        @Override
+        public boolean isEmpty() {
+            return delegate.isEmpty();
+        }
+
+        @Override
+        public Iterator<T> iterator() {
+            return delegate.iterator();
+        }
+
+        @Override
+        public int lastIndexOf(Object obj) {
+            return narrowed().lastIndexOf(obj);
+        }
+
+        @Override
+        @SuppressWarnings("unchecked")
+        public ListIterator<T> listIterator() {
+            return listIteratorFactory.apply((S) delegate, 0);
+        }
+
+        @Override
+        @SuppressWarnings("unchecked")
+        public ListIterator<T> listIterator(int index) {
+            return listIteratorFactory.apply((S) delegate, index);
+        }
+
+        @Override
+        public T remove(int index) {
+            // may throw an IndexOutOfBoundsException accordingly to the j.u.List.remove(int)
+            return setDelegateAndGetPreviousElement(index, delegate.removeAt(index));
+        }
+
+        @Override
+        @SuppressWarnings("unchecked")
+        public boolean remove(Object obj) {
+            return setDelegateAndCheckChanged(narrowed().remove(obj));
+        }
+
+        public Seq<Object> narrowed() {
+            return Seq.narrow(delegate);
+        }
+
+        @Override
+        public boolean removeAll(Collection<?> c) {
+            return setDelegateAndCheckChanged(narrowed().removeAll(c));
+        }
+
+        @Override
+        public boolean retainAll(Collection<?> c) {
+            return setDelegateAndCheckChanged(narrowed().retainAll(c));
+        }
+
+        @Override
+        public T set(int index, T element) {
+            // may throw an IndexOutOfBoundsException accordingly to the j.u.List.set(int, T)
+            return setDelegateAndGetPreviousElement(index, delegate.update(index, element));
+        }
+
+        @Override
+        public int size() {
+            return delegate.size();
+        }
+
+        @Override
+        public java.util.List<T> subList(int fromIndex, int toIndex) {
+            // may throw IndexOutOfBoundsException accordingly to j.u.List.subList(int, int)
+            return delegate.subSequence(fromIndex, toIndex).asJava();
+        }
+
+        @Override
+        public Object[] toArray() {
+            return delegate.toJavaArray();
+        }
+
+        @Override
+        public <U> U[] toArray(U[] array) {
+            return delegate.toJavaList().toArray(array);
+        }
+
+        // -- Object.*
+
+        @Override
+        public boolean equals(Object o) {
+            return (o == this) || ((o instanceof java.util.List) && areEqual(delegate, (java.util.List<?>) o));
+        }
+
+        @Override
+        public int hashCode() {
+            return delegate.hashCode();
```

Comment:
Is this hashCode consistent with equals? Java collections break so many rules it's hard to keep track.

Going by the javadocs for AbstractList and List.hashCode, it might be best to call AbstractList.hashCode.